packages/adapters/bitcoin/src/connectors/LeatherConnector.ts

```diff
@@ -0,0 +1,74 @@
+import type { BitcoinConnector } from '../index.js'
+import { SatsConnectConnector } from './SatsConnectConnector.js'
+import type {
+  SendTransferRequestParams,
+  SendTransferResponseBody,
+  SignPsbtRequestParams,
+  SignPsbtResponseBody
+} from '@leather.io/rpc'
+
+export class LeatherConnector extends SatsConnectConnector {
+  static readonly ProviderId = 'LeatherProvider'
+
+  constructor({ connector }: LeatherConnector.ConstructorParams) {
+    if (connector.wallet.id !== LeatherConnector.ProviderId) {
+      throw new Error('LeatherConnector: wallet must be a LeatherProvider')
+    }
+
+    super({
+      provider: connector.wallet,
+      requestedChains: connector.requestedChains
+    })
+  }
+
+  public override async sendTransfer({
+    amount,
+    recipient
+  }: BitcoinConnector.SendTransferParams): Promise<string> {
+    const params: LeatherConnector.SendTransferParams = {
+      address: recipient,
+      amount
+    }
+
+    const res: LeatherConnector.SendTransferResponse = await this.internalRequest(
+      'sendTransfer',
+      // @ts-expect-error - expected LeatherWallet params don't match sats-connect
+      params
+    )
+
+    return res.txid
+  }
+
+  public override async signPSBT({
+    psbt,
+    broadcast = false
+  }: BitcoinConnector.SignPSBTParams): Promise<BitcoinConnector.SignPSBTResponse> {
+    const params: LeatherConnector.SignPSBTParams = {
+      hex: Buffer.from(psbt, 'base64').toString('hex'),
+      network: 'mainnet',
+      broadcast
+    }
+
+    // @ts-expect-error - expected LeatherWallet params don't match sats-connect
+    const res: LeatherConnector.SignPSBTResponse = await this.internalRequest('signPsbt', params)
+
+    return {
+      psbt: Buffer.from(res.hex, 'hex').toString('base64'),
+      txid: res.txid
+    }
+  }
+}
+
+export namespace LeatherConnector {
+  export type ConstructorParams = { connector: SatsConnectConnector }
+
+  export type Network = 'mainnet' | 'testnet' | 'signet' | 'sbtcDevenv' | 'devnet'
```

Comment:
should we make this a constant?

Reply:
I don't think we need to add a constant for this as it is specific typing of LeatherConnector.